account-kit/rn-signer/src/client.ts

```diff
@@ -116,15 +148,61 @@ export class RNSignerClient extends BaseSignerClient<undefined> {
     this.eventEmitter.emit(params.connectedEventName, user, params.bundle);
     return user;
   }
-  override oauthWithRedirect(
+  override oauthWithRedirect = async (
     _args: Extract<OauthParams, { mode: "redirect" }>
-  ): Promise<never> {
-    throw new Error("Method not implemented.");
-  }
+  ): Promise<User> => {
+    try {
+      const providerUrl = await this.getOauthProviderUrl(_args);
+      const redirectUrl = _args.redirectUrl;
+
+      return new Promise(async (resolve, reject) => {
+        if (await InAppBrowser.isAvailable()) {
+          const res = await InAppBrowser.openAuth(providerUrl, redirectUrl);
+
+          if (res.type !== "success" || !res.url) {
+            return reject(
+              new OauthFailedError("An error occured completing your request")
+            );
+          }
+
+          const authResult = parseSearchParams(res.url);
+          const bundle = authResult["alchemy-bundle"] ?? "";
+          const orgId = authResult["alchemy-org-id"] ?? "";
+          const idToken = authResult["alchemy-id-token"] ?? "";
+          const isSignup = authResult["alchemy-is-signup"];
+          const error = authResult["alchemy-error"];
+
+          if (bundle && orgId && idToken) {
+            this.completeAuthWithBundle({
+              bundle,
+              orgId,
+              connectedEventName: "connectedOauth",
+              idToken,
+              authenticatingType: "oauth",
+            }).then((user) => {
+              if (isSignup) {
+                this.eventEmitter.emit("newUserSignup");
+              }
+              resolve(user);
+            }, reject);
+          } else if (error) {
+            reject(new OauthFailedError(error));
+          }
+        } else {
+          throw new Error(
+            "Couldn't perfom authentication operation. Browser not available."
+          );
```

Comment:
nit: let's add an `errors.ts` file in `rn-signer/src` and in it define strongly typed errors that extends `BaseError` exported from `aa-sdk/core`.

That will allow us to link to specific docs based on the error being thrown